app/crud/microservice.py

```diff
@@ -1,17 +1,72 @@
-
+import sqlalchemy
 from sqlalchemy.orm import Session
-
-from ..models import Microservice
-from ..schemas import MicroserviceCreate, MicroserviceUpdate
+from fastapi.responses import JSONResponse 
+from ..models import Microservice ,Team ,Scorecard ,MicroserviceScoreCard
+from ..schemas import MicroserviceCreate, MicroserviceUpdate ,MicroserviceInDBBase , TeamInDBBase ,ScoreCardInDBBase
 from .base import CRUDBase
+from typing import List
+from sqlalchemy.sql import func
+from starlette.exceptions import HTTPException
+
 
 
-class CRUDMicroservice(CRUDBase[Microservice, MicroserviceCreate, MicroserviceUpdate]):
+
+class CRUDMicroservice(CRUDBase[Microservice, MicroserviceCreate, MicroserviceUpdate ]):
     def __init__(self, db_session: Session):
         super(CRUDMicroservice, self).__init__(Microservice, db_session)
-
+        #self.teamService= CRUDTeam(db_session)
+        #self.scoreCardService= CRUDMicroserviceScoreCard(db_session)
+        
     def getByTeamId(self, teamId: str):
         return self.db_session.query(Microservice).filter(Microservice.teamId == teamId).all()
 
     def getByTeamIdAndCode(self, teamId: str, code: str):
         return self.db_session.query(Microservice).filter(Microservice.teamId == teamId, Microservice.code == code).first()
+    
+    #get all with team
+    #def getAllByTeamName(self) -> list[MicroserviceInDBBase]:
+    #    result = (
+    #    self.db_session.query(Microservice.id, Microservice.name, Microservice.description, Microservice.code, Team.name.label("team_name"))
+    #    .join(Team, Microservice.teamId == Team.id).all()
+    #    )
+    #    return result
+    
+    def getAllServicesByTeamName(self) -> list[MicroserviceInDBBase]:
+        
+        microservices = self.list()
+        services = []
+        
+        for microservice in microservices:
+        # Get the related team for each microservice
+            team = self.db_session.query(Team).filter(Team.id == microservice.teamId).first()
+            
+            service = MicroserviceInDBBase(
+                id=microservice.id,
+                name=microservice.name,
+                description=microservice.description,
+                code=microservice.code,
+                team_name=team.name if team else None,
+            )
+            services.append(service)
+         
+        return services
+   
+   #get one with team
+    def getByServiceIdWithTeamName(self , service_id:int):
+        result = (
+        self.db_session.query(Microservice.id, Microservice.name, Microservice.description, Microservice.code, Team.name.label("team_name"))
+        .join(Team, Microservice.teamId == Team.id)
```

Comment:
joining two tables is a costly operation, you shouldn't do it If you can obtain the information you want without joining tables.
And in this case, you can get all fields of the team without joining the two tables. You can query the Microservice table and get the record which includes the teamId, then you can use this teamId to query the Team table to get the team record. Then inject the team object in the micorservice object and return the result.
Query by Id is a fast operation, it benefits from DB tables indexing.

Please search this further and improve your code or prove that your current implementation is more efficient than the proposed one : ).